### PR TITLE
OKD4 (UPI) on Openstack with terraform

### PR DESCRIPTION
**Pull request by JarleB**

### CODE REVIEW

File: Guides/UPI/okd4-terraform-openstack/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Openshift community OKD4 (UPI) on Openstack with terraform
+
+Terraform code to provision OKD4 using user provisioned installer (UPI) and
+glue code to tie it all together
+
+From an openshift installer perspective this is a "bare metal" install:
+https://docs.okd.io/latest/installing/installing_bare_metal/installing-bare-metal.html#installing-bare-metal
+User provided infrastructure is orchestrated using terraform in combination with scripts and an ansible playbook.
+
+Tested on Openstack RDO with Calico as SDN in Openstack and openstack designate
```
Comment (vrutkovs) [inline Guides/UPI/okd4-terraform-openstack/README.md:10]: Interesting, I heard Kuryr would integrate into OpenStack better. Any particular reason to pick Calico?

File: Guides/UPI/okd4-terraform-openstack/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+for DNS setup within terraform (specifically https://nrec.no/)
+
+Disclaimer: This code is for testing / proof-of-concept and not for production use.
+
+NB: You don't want to be logged in to other openshift cluster contexts when
+running this deployment. It might take you in an unexpected direction :-)
+
+## How to use
+
+First clone this repo to local host and `cd okd4-terraform-openstack`
```
Comment (vrutkovs) [inline Guides/UPI/okd4-terraform-openstack/README.md:20]:
```suggestion
First clone this repo to local host and `cd Guides/UPI/okd4-terraform-openstack`
```

File: Guides/UPI/okd4-terraform-openstack/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+watch ./oc get co
+```
+
+Finally csr's of worker nodes are approved, (or actually all pending csr's are
+approved, assuming it is the nodes that requested them, Be aware)
+
+When all cluster operators are ready, you should be able to log in to the web
+console. To find the address:
+
+` ./oc get route --all-namespaces |grep ' console'`
```
Comment (vrutkovs) [inline Guides/UPI/okd4-terraform-openstack/README.md:112]: Use `wait-for install-complete` in the installer? This should output console URL and kubeadmin password afaik

File: Guides/UPI/okd4-terraform-openstack/scripts/update-installer-and-image.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+repo="registry.svc.ci.openshift.org/origin/release"
+if  [[ -z $2 ]]
+then
+  echo "Usage: $0 <okd-release> <fcos-image-version>"
+  exit 2
+fi
+
+release="$1"
+image_ver="${2}.0"
+fcos_repo="https://builds.coreos.fedoraproject.org/prod/streams/testing-devel/builds"
```
Comment (vrutkovs) [inline Guides/UPI/okd4-terraform-openstack/scripts/update-installer-and-image.sh:21]: Generally we recommend starting with stable releases, but that would work too

File: Guides/UPI/okd4-terraform-openstack/README.md (hunk 1 above)
Comment (JarleB) [reply Guides/UPI/okd4-terraform-openstack/README.md:10]: Just that this is what ww have at han, and nrec.no was designed some years back and is used for a lot of other stuff than okd. It is not an option ( at the moment  at least) to swap sdn to something else at the moment. We have to deal with what is there.

File: Guides/UPI/okd4-terraform-openstack/README.md (hunk 3 above)
Comment (JarleB) [reply Guides/UPI/okd4-terraform-openstack/README.md:112]: Thanks, I'll change it.

**Comment by JarleB** [comment]
I think all was covered in subsequent commits? Anything else?

File: Guides/UPI/okd4-terraform-openstack/README.md (hunk 2 above)
Comment (vrutkovs) [reply Guides/UPI/okd4-terraform-openstack/README.md:20]: The latest commit switched it back :(

**Comment by vrutkovs** [comment]
Looks good, lets fix one last path issue, squash - and its good to go.

Thanks!

**Comment by JarleB** [comment]
Like so?

File: Guides/UPI/okd4-terraform-openstack/README.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+## How to use
+
+First clone this repo to local host and `cd Guides/UPI/okd4-terraform-openstack`
+
+There is two scripts, `update-installer-and-image.sh` and
+`deploy.sh`, for getting dependencies and deploying the
+cluster respectively.
+
+You need access to an openstack project with enoght resources to accomodate nodes according to
```
Comment (lhoss) [inline Guides/UPI/okd4-terraform-openstack/README.md:26]: typo: enoght

File: Guides/UPI/okd4-terraform-openstack/README.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+In order to use the update-installer-and-image.sh you must first be
+authenticated to the openstack project (in order to upload the FCOS image).
+
+Then you can for example do:
+
+`./scripts/update-installer-and-image.sh 4.4.0-0.okd-2020-04-21-163702-beta4 31.20200420.20`
+
+It is of course also possible to combine any version of FCOS and OKD and/or
+download directly from the realease pages of OKD and Fedora CoreOS and manually
```
Comment (lhoss) [inline Guides/UPI/okd4-terraform-openstack/README.md:46]: typo: realease

File: Guides/UPI/okd4-terraform-openstack/README.md (hunk 5 above)
Comment (vrutkovs) [reply Guides/UPI/okd4-terraform-openstack/README.md:26]: Oh, could you create a PR with fixes for that?

**Pull request merged**